version: "3.8"
services:
  web:
    build:
      context: app/fakt-news
      args:
        - USER=${PY_USER}
        - USER_ID=${PY_USER_ID}
        - GROUP_ID=${PY_GROUP_ID}
    env_file: .env
    ports:
      - 3000:3000
    volumes: 
      - "./app/fakt-news:/app"
    depends_on:
      - "rdf"

  python:
    build:
      context: data
      args:
        - USER=${PY_USER}
        - USER_ID=${PY_USER_ID}
        - GROUP_ID=${PY_GROUP_ID}

    env_file: .env
    volumes:
      - "./data:/app" #for code sync
      - "./ontology:/app/rdf" #for code sync
    depends_on:
      - "db"

  db:
    image: postgres:13.1
    restart: always
    env_file: .env
    ports:
      - 5432:5432
    volumes:
      - db-volume:/var/lib/postgresql/data

  rdf:
    image: ontotext/graphdb:9.6.0-free # is build locally!!
    ports:
      - 7200:7200
    volumes:
      - rdf-volume:/opt/graphdb/home
    #  - ./license:/opt/graphdb/home/conf # for graphdb se license

  # react:
  #   build:
  #     context: app/fakt-news
  #   ports:
  #     - 3001:3000
  #   volumes:
  #     - "./app/fakt-news:/app" # for code sync
  #     - "/app/node_modules/" # for code sync

volumes:
  db-volume:
  rdf-volume:
